zerver/actions/streams.py

```diff
@@ -74,6 +74,38 @@
 from zerver.tornado.django_api import send_event, send_event_on_commit
 
 
+def send_user_remove_events_on_stream_deactivation(
+    stream: Stream, subscribed_users: List[UserProfile]
+) -> None:
+    guest_subscribed_users = [user for user in subscribed_users if user.is_guest]
```

Comment:
Should this exit early if `guest_subscribed_users` was empty?

Reply:
Added such a check. It's not important given that stream deactivations are rare and expensive, though.